mlflow/resource_mlflow_model.go

```diff
@@ -3,112 +3,78 @@ package mlflow
 )
 
-type Tag struct {
-	Key   string `json:"key"`
-	Value string `json:"value"`
-}
-
-// ModelVersion defines a MLFlow model version as returned by the API
-type ModelVersion struct {
-	Name                 string `json:"name"`
-	Version              string `json:"version"`
-	CreationTimestamp    int64  `json:"creation_timestamp,omitempty"`
-	LastUpdatedTimestamp int64  `json:"last_updated_timestamp,omitempty"`
-	UserID               string `json:"user_id,omitempty"`
-	CurrentStage         string `json:"current_stage,omitempty"`
-	Source               string `json:"source,omitempty"`
-	Status               string `json:"status,omitempty"`
-}
-
-// Model defines the response object from the API
-type Model struct {
-	Name                 string         `json:"name" tf:"force_new"`
-	CreationTimestamp    int64          `json:"creation_timestamp,omitempty" tf:"computed"`
-	LastUpdatedTimestamp int64          `json:"last_updated_timestamp,omitempty" tf:"computed"`
-	UserID               string         `json:"user_id,omitempty" tf:"computed"`
-	LatestVersions       []ModelVersion `json:"latest_versions,omitempty" tf:"computed"`
-	Description          string         `json:"description,omitempty"`
-	Tags                 []Tag          `json:"tags,omitempty"`
-	RegisteredModelID    string         `json:"id,omitempty" tf:"computed,alias:registered_model_id"`
-}
-
-// registeredModel defines response from GET API op
-type registeredModel struct {
-	RegisteredModelDatabricks Model `json:"registered_model_databricks"`
-}
-
-type ModelsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewModelsAPI(ctx context.Context, m any) ModelsAPI {
-	return ModelsAPI{m.(*common.DatabricksClient), ctx}
-}
-
-func (a ModelsAPI) Create(m *Model) error {
-	return a.client.Post(a.context, "/mlflow/registered-models/create", m, m)
-}
-
-func (a ModelsAPI) Read(name string) (*Model, error) {
-	var m registeredModel
-	err := a.client.Get(a.context, "/mlflow/databricks/registered-models/get", map[string]string{
-		"name": name,
-	}, &m)
-	if err != nil {
-		return nil, err
-	}
-	return &m.RegisteredModelDatabricks, nil
-}
-
-// Update the model entity
-func (a ModelsAPI) Update(m *Model) error {
-	return a.client.Patch(a.context, "/mlflow/registered-models/update", m)
-}
-
-// Delete removes the model by its name
-func (a ModelsAPI) Delete(name string) error {
-	return a.client.Delete(a.context, "/mlflow/registered-models/delete", map[string]string{
-		"name": name,
-	})
-}
-
 func ResourceMlflowModel() *schema.Resource {
 	s := common.StructToSchema(
-		Model{},
-		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			delete(m, "latest_versions")
-			return m
+		ml.Model{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			delete(s, "latest_versions")
+			s["name"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			}
```

Comment:
The schema should already be `TypeString` if I'm not mistaken. This will ensure that customizations otherwise propagated to the `Schema` of this field will not be overwritten.
```suggestion
			s["name"].Required = true
			s["name"].ForceNew = true
```

Reply:
This commit throws an error. Also specifying a schema without a type throws an error. Will leave as is unless you suggest otherwise

Reply:
Ok. I maybe misunderstood as I thought the name field would have been filled in by StructToSchema already, so the point was to only change the attributes of the schema you care about rather than replacing the entire schema of this field. However, given that this is a string field and these are its only attributes, I feel ok with this as well.